sockeye/inference.py

```diff
@@ -195,28 +205,47 @@ def sym_gen(bucket_key: Tuple[int, int]):
             Returns either softmax output (probs over target vocabulary) or inputs to logit
             computation, controlled by decoder_return_logit_inputs
             """
-            source_max_len, target_max_len = bucket_key
-            source_encoded_seq_len = self.encoder.get_encoded_seq_len(source_max_len)
+            source_seq_len, target_seq_len = bucket_key
+            source_embed_seq_len = self.embedding_source.get_encoded_seq_len(source_seq_len)
+            source_encoded_seq_len = self.encoder.get_encoded_seq_len(source_embed_seq_len)
 
             self.decoder.reset()
-            prev_word_ids = mx.sym.Variable(C.TARGET_NAME)
+            target = mx.sym.Variable(C.TARGET_NAME)
+            target_length = utils.compute_lengths(target)
```

Comment:
minor: I guess it would make sense to use the plural `target_lengths` here as well.

Reply:
will fix